packages/arb-token-bridge-ui/src/util/deposits/helpers.ts

```diff
@@ -123,7 +130,37 @@ const updateAdditionalDepositDataETH = async ({
   return updatedDepositTx
 }
 
-const updateAdditionalDepositDataToken = async ({
+const getTokenDepositStatusData = async ({
+  l1ToL2Msg
+}: {
+  l1ToL2Msg: L1ToL2MessageReader
+}): Promise<L1ToL2MsgData> => {
+  // fetch basic status of token deposit
+  const status = await l1ToL2Msg.status()
+  const l1ToL2MsgData: L1ToL2MsgData = {
+    status: status,
+    l2TxID: undefined,
+    fetchingUpdate: false,
+    retryableCreationTxID: l1ToL2Msg.retryableCreationId
+  }
+
+  // if status is REDEEMED, only then fetch the L2TxId (`waitForStatus` waits for tx information until redeemed)
+  // if this REDEEMED check isn't here, the `await` will not get resolved for a long time!
+  if (status === L1ToL2MessageStatus.REDEEMED) {
+    // if `waitForStatus` is called on unredeemed tx, it waits for it to get redeemed first (~15mins)
+    const res = await l1ToL2Msg.waitForStatus()
```

Comment:
Can we use `getSuccessfulRedeem` here?

Reply:
Good point. Used this in both the places where `waitForStatus` was called, and minimised the diff as well.